fuel/datasets/hdf5.py

```diff
@@ -64,3 +67,127 @@ def get_data(self, state=None, request=None):
                 raise ValueError
         data = [node[request] for node in self.nodes]
         return data
+
+
+@do_not_pickle_attributes('data_sources')
+class H5PYDataset(Dataset):
+    """An h5py-fueled HDF5 dataset.
+
+    This dataset class assumes a particular file layout:
+
+    * Data sources reside in the root group, and their names define the
+      source names.
+    * The dataset is not explicitly split. Instead, splits are defined as
```

Comment:
The corpora argument basically just comes down to this: WMT15 data consists of several corpora, and even the test set consists of multiple different corpora. It would be nice if you can do `WMT15('europarl')`, `WMT15('newstest1213')`, `WMT15('test')`, etc. Here `WMT15('test')` is a combination of `WMT15('newstest1213')` as well as `WMT15('newstest14')`, but depending on whether you're combining with an SMT system and need a separate set to tune, you might want to use either `test` or `newstest1213` as your test set.